dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -0,0 +1,249 @@
+		
+		DotConnect dc = new DotConnect();
+		
+		//Get all the virtual_links
+		dc.setSQL(SELECT_VIRTUAL_LINK_QUERY);
+		List<Map<String,String>> virtualLinks = dc.loadResults();
+		
+		//Update Inode Table, since previously the type was 'virtual_link' and now needs to be 'contentlet'
+		dc.setSQL(UPDATE_INODE_QUERY);
+		dc.loadResult();
+		Logger.info(getClass(), "Type in the Inode Table has been changed to contentlet");
+		
+		// Deletes all rows in the Virtual Link Table
+		dc.setSQL(DELETE_VIRTUAL_LINK_QUERY);
+		dc.loadResult();
+		
+		for (Map<String, String> virtualLink : virtualLinks) {
+
+			String oldURL = virtualLink.get("url").toString();
+			String inode = virtualLink.get("inode").toString();
+			String title = virtualLink.get("title").toString();
+			String oldUri = virtualLink.get("uri").toString();
+
+			// Calls the method that splits the URL into the site and the uri
+			Map<String, String> siteAndUriMap = splitURL(oldURL);
+
+			String site = siteAndUriMap.get("siteId");
+			String uri = siteAndUriMap.get("uri");
+
+			//If the site returns null means that there is no host
+			if (site == null) {
+				Logger.warn(this,
+						"Vanity URL cannot be migrated to Content because the Host does not exist. You will need to create it manually, Title: "
+								+ title + ", URL: " + oldURL + ", URI: " + oldUri);
+				dc.setSQL(DELETE_INODE_BY_INODE_QUERY);
+				dc.addParam(inode);
+				dc.loadResult();
+			} else {
+
+				String uuid = UUIDGenerator.generateUuid();
+				
+				int lang = Integer.parseInt(findHostLang(site)); //Gets the lang of the Site, so when saving the Vanity URL it's the same lang
+
+				// Insert on Identifier Table
+				dc.setSQL(INSERT_IDENTIFIER_QUERY);
+				dc.addParam(uuid); // identifier
+				dc.addParam("/"); // parent_path
+				dc.addParam("content." + inode); // asset_name
+				dc.addParam(Host.SYSTEM_HOST); // host_inode
+				dc.addParam("contentlet"); // asset_type
+				dc.loadResult();
+
+				// Insert on Contentlet Table
+		        if (DbConnectionFactory.isMySql()) {
+		            // Use correct escape char when using reserved words as column names
+		        	INSERT_CONTENTLET_QUERY = INSERT_CONTENTLET_QUERY.replaceAll("\"", "`");
+		        }
+				dc.setSQL(INSERT_CONTENTLET_QUERY);
+				dc.addParam(inode); // inode
+				dc.addParam(false); // show_on_menu
+				dc.addParam(0); // sort_order
+				dc.addParam(title); // title
+				dc.addParam(new Date()); // mod_date
+				dc.addParam("dotcms.org.1"); // mod_user
+				dc.addParam(title); // friendly_name
+				dc.addParam("8e850645-bb92-4fda-a765-e67063a59be0"); // structure_inode
+				dc.addParam(uuid); // identifier
+				dc.addParam(lang); // language_id
+				dc.addParam(title); // text1(title)
+				dc.addParam(site); // text2(site)
+				dc.addParam(uri); // text3(uri)
+				dc.addParam(oldUri); // text4(forwardTo)
+				dc.addParam(301); // integer1(action)
+				dc.addParam(0); // integer2(sort)
+				for (int i = 0; i < 48; i++) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 48 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)